src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPEditorImprovementsTypedHandler.java

```diff
@@ -151,14 +168,18 @@ private static boolean inStringLiteral(@NotNull PsiFile file,
     private static boolean inComment(@NotNull PsiFile file,
                                      @NotNull CharSequence documentChars,
                                      int offset) {
-        boolean inComment = false;
+        // Try to determine this using semantic tokens
+        LSPSemanticTokensFileViewProvider semanticTokensFileViewProvider = LSPSemanticTokensFileViewProvider.getInstance(file);
+        if ((semanticTokensFileViewProvider != null) && semanticTokensFileViewProvider.isComment(offset)) {
+            return true;
+        }
 
         Document document = LSPIJUtils.getDocument(file);
         if (document != null) {
             Commenter commenter = LSPIJEditorUtils.getCommenter(file);
 
-            // First try to see if we're in a line comment since that's much more efficient. We're in a line comment if
-            // we find a line comment start on the same line before the current position.
+            // Try to see if we're in a line comment since that's much more efficient. We're in a line comment if we
```

Comment:
Removed the unnecesary `inComment` flag and return `true` immediately when we know that we're in a comment.